src/io/input/xml/XMLFileReader.cpp

```diff
@@ -3,26 +3,85 @@
 #include <optional>
 
 #include "io/logger/Logger.h"
+#include "io/particle_spawners/CuboidSpawner.h"
 
-SimulationParams XMLFileReader::readConfiguration(const std::string& filepath, ParticleContainer& particle_container) const {
-    if (filepath.empty()) {
-        Logger::logger->error("No input file specified.");
-        exit(-1);
-    }
-
+SimulationParams XMLFileReader::readFile(const std::string& filepath, ParticleContainer& particle_container) const {
     try {
         std::unique_ptr< ::configuration> config = configuration_(filepath);
 
-        Logger::logger->info("Successfully parsed XML file.");
+        Logger::logger->info("Successfully parsed XML file.Simulation parameters: ");
         Logger::logger->info("FPS: {}", config->fps().get());
+        Logger::logger->info("Video length: {}", config->video_length().get());
+        Logger::logger->info("End time: {}", config->end_time().get());
+        Logger::logger->info("Delta t: {}", config->delta_t().get());
+
+        if(config->fps().get() < 0){
+            Logger::logger->error("FPS must be positive");
+            exit(-1);
+        }
+        if(config->video_length().get() < 0){
+            Logger::logger->error("Video length must be positive");
+            exit(-1);
+        }
+        if(config->end_time().get() < 0){
+            Logger::logger->error("End time must be positive");
+            exit(-1);
+        }
+        if(config->delta_t().get() < 0){
+            Logger::logger->error("Delta t must be positive");
+            exit(-1);
+        }
+
+        const std::string& output_dir_path = "";
+        const std::string& log_level = "";
+        auto params = SimulationParams(filepath, output_dir_path, config->delta_t().get(), config->end_time().get(),
+                                       config->fps().get(), config->video_length().get(), log_level);
+
+        for(int i = 0;i<config->cuboid().size();i++){
+            std::array<double, 3> lower_left_front_corner{config->cuboid()[i].position().x(), config->cuboid()[i].position().y(), config->cuboid()[i].position().z()};
+            if (lower_left_front_corner[0] < 0 || lower_left_front_corner[1] < 0 || lower_left_front_corner[2] < 0){
+                Logger::logger->error("Cuboid position must be positive");
+                exit(-1);
+            }
+            std::array<long long int, 3> grid_dimensions{config->cuboid()[i].grid_dim().dimx(), config->cuboid()[i].grid_dim().dimy(), config->cuboid()[i].grid_dim().dimz()};
+            if(grid_dimensions[0] < 0 || grid_dimensions[1] < 0 || grid_dimensions[2] < 0){
+                Logger::logger->error("Cuboid grid dimensions must be positive");
+                exit(-1);
+            }
+            if(grid_dimensions[0] > 2147483647 || grid_dimensions[1] > 2147483647 || grid_dimensions[2] > 2147483647){
```

Comment:
are those overflow checks even necessary?

couldnt we just restrict the range using the xml-types from the schema?

There exists some xml-datatypes that exactly match to c++ types:  so we dont need to check the range afterwards: https://www.codesynthesis.com/projects/xsd/documentation/cxx/tree/manual/#2.5

Reply:
fixed this in 8637d42